tests/e2e/components_test.go

```diff
@@ -0,0 +1,280 @@
+package e2e_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/apis/common"
+	dscv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/datasciencecluster/v1"
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/utils/test/matchers/jq"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/utils/test/testf"
+
+	. "github.com/onsi/gomega"
+)
+
+type ComponentTestCtx struct {
+	*testf.TestContext
+
+	GVK                  schema.GroupVersionKind
+	DSCName              types.NamespacedName
+	DSCIName             types.NamespacedName
+	ApplicationNamespace string
+}
+
+func NewComponentTestCtx(object common.PlatformObject) (*ComponentTestCtx, error) {
+	tcf, err := testf.NewTestContext(
+		testf.WithTOptions(
+			testf.WithEventuallyTimeout(2*time.Minute),
```

Comment:
Should this magic numbers (and 30 secs below) be moved somewhere out to constant section? I do not mean it should be configured from outside.

Reply:
done